pkg/plugins/secrets/k8s/store.go

```diff
@@ -119,6 +119,9 @@ func (s *KubernetesStore) Get(ctx context.Context, r *secret_model.SecretResourc
 	if err := s.converter.ToCoreResource(secret, r); err != nil {
 		return errors.Wrap(err, "failed to convert k8s Secret into core counterpart")
 	}
+	if r.GetMeta().GetMesh() != opts.Mesh {
+		return core_store.ErrorResourceNotFound(r.GetType(), opts.Name, opts.Mesh)
+	}
```

Comment:
Interesting, this will still return the valid resource in `r`. When we talk secrets we'd better be careful, shall we fetch the secret in a local variable and copy it only when everything is successful?
Also this mesh filtering becomes more and more a problem, let's think about the labels for filtering on the K8s server.

Reply:
That's a really good catch, thank you.

I struggle with the mutability of this language...
Let me create issue with proposal about migrating from `Mesh: default` to label `kuma.io/mesh: default`